### PR TITLE
make filling chars (and, thus, erase line/char) unset wrap

### LINKED ISSUE #1126: Copying text from the command window behaves differently than previous versions
Ported from internal bug
>When I select some text from a window using Vim.exe and paste to another window, the text is badly formatted.
>Also, if double-click text from a command window, it copies/pastes the extra white-space at the end which was not the case in earlier releases.
>I also see the problem when copy/pasting to an excel spreadsheel

Tried to repro: space bug does not repro, vim bug does since 19H1

### PR DESCRIPTION
**Pull request by carlos-zamora**
## Summary of the Pull Request

EraseInLine calls `FillConsoleOutputCharacterW()`. In filling the row with chars, we were setting the wrap flag. We need to specifically not do this on ANY _FILL_ operation. Now a fill operation UNSETS the wrap flag if we fill to the end of the line.

## PR Checklist
* [x] Closes #1126 
* [x] CLA signed.
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [x] I am a core contributor

## Detailed Description of the Pull Request / Additional comments

Originally, we had a boolean `setWrap` that would mean...
- **true**: if writing to the end of the row, SET the wrap value to true
- **false**: if writing to the end of the row, DON'T CHANGE the wrap value

Now we're making this bool a std::optional to allow for a ternary state. This allows for us to handle the following cases completely. Refer to the table below:

| current wrap value | are we filling the last cell in the row? | new wrap value | COMMENTS |
|-- |-- |-- |-- |
| 0 | 0 | 0 | |
| ~0~ | ~0~ | ~1~ | IMPOSSIBLE |
| 0 | 1 | 0 | |
| 0 | 1 | 1 | THIS CASE WAS RIGHTFULLY HANDLED |
| 1 | 0 | 0 | THIS CASE WAS UNHANDLED |
| 1 | 0 | 1 | |
| ~1~ | ~1~ | ~0~ | IMPOSSIBLE |
| 1 | 1 | 1 | |

To handle that special case (1-0-0), we need to UNSET the wrap. So now, we have ~setWrap~ `wrap` mean the following:
- **true**: if writing to the end of the row, SET the wrap value to TRUE
- **false**: if writing to the end of the row, SET the wrap value to FALSE
- **nullopt**: leave the wrap value as it is

## Validation Steps Performed
I feel like we should add a test for this case. But idk where or how. Help?

### CODE REVIEW

File: src/buffer/out/Row.cpp [hunk 1]
```diff
@@ -202,10 +202,15 @@ OutputCellIterator ROW::WriteCells(OutputCellIterator it, const size_t index, co
                 ++it;
             }
 
-            // If we're asked to set the wrap status and we just filled the last column with some text, set wrap status on the row.
-            if (setWrap && fillingLastColumn)
+            // If we're asked to (un)set the wrap status and we just filled the last column with some text...
+            // NOTE:
+            //  - wrap = std::nullopt    --> we don't care about changing the value
```
Comment (miniksa) [inline src/buffer/out/Row.cpp]: FYI, I figured out when this is applicable (at least one case).

It's when we use the `Popup` class to show an overlay for CMD (like listing out the most recently used commands). We don't want the wrap status messed with during either the backup or restore. 

I think I verbally said that I wasn't sure of a circumstance where the `std::nullopt` case would be used. But now I am.

File: src/host/_output.cpp [hunk 2]
```diff
@@ -290,7 +290,10 @@ void WriteToScreen(SCREEN_INFORMATION& screenInfo, const Viewport& region)
     try
     {
         const OutputCellIterator it(character, lengthToWrite);
-        const auto done = screenInfo.Write(it, startingCoordinate);
+
+        // when writing to the buffer, specifically unset wrap if we get to the last line.
```
Comment (miniksa) [inline src/host/_output.cpp]:
```suggestion
        // when writing to the buffer, specifically unset wrap if we get to the last column.
```

File: src/buffer/out/Row.cpp (hunk 1 above)
Comment (carlos-zamora) [reply src/buffer/out/Row.cpp]: In that case, I'm specifically changing the comment to "don't change the wrap value" for clarity.

**Comment by miniksa** [comment]
> @miniksa Michael Niksa FTE mind looking at these new tests? They seem sound to me.

Sure.

File: src/host/ft_host/API_FillOutputTests.cpp [hunk 3]
```diff
@@ -58,4 +58,140 @@ class FillOutputTests
...
+        // Verify wrap occurred
+        std::unique_ptr<wchar_t[]> bufferText = std::make_unique<wchar_t[]>(ogConsoleWidth);
+        DWORD readSize = 0;
+        VERIFY_WIN32_BOOL_SUCCEEDED(ReadConsoleOutputCharacterW(hConsole,
+                                                                bufferText.get(),
+                                                                ogConsoleWidth,
+                                                                { 0, 0 },
+                                                                &readSize));
+
+        for (DWORD i = 0; i < readSize; i++)
```
Comment (miniksa) [inline src/host/ft_host/API_FillOutputTests.cpp]: Man.... can you do some sort of string compare here instead?

`WEX::Common::String()` has a bunch of helpers for allocating two strings that are comparable in `VERIFY_ARE_EQUAL` and show up in the logs correctly.

**Pull request merged**